Source/Session/Session.swift

```diff
@@ -225,7 +229,7 @@ extension Session: VisitableDelegate {
         } else if visitable === currentVisit.visitable && currentVisit.state == .started {
             // Navigating forward - complete navigation early
             completeNavigationForCurrentVisit()
-        } else if visitable !== topmostVisit.visitable {
+        } else if visitable !== topmostVisit.visitable || visitable === lastDisappearingVisit?.visitable {
```

Comment:
This covers the case of restoring from native screen back to web, which previously wasn't considered a restore. The web view would both topmost and current visit, even when navigating back.

Reply:
This works as expected.
Since this is an or operation I'd propose to evaluate the latter in a separate `else if` statement.
This will make it clear that we're dealing with two types of backward navigation.
I've addressed it and added comments in my feedback PR https://github.com/hotwired/turbo-ios/pull/189